packages/main/src/SplitButton.js

```diff
@@ -0,0 +1,409 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import {
+	isEscape,
+	isSpace,
+	isEnter,
+	isDown,
+	isUp,
+	isDownAlt,
+	isUpAlt,
+	isF4,
+	isShift,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import SplitButtonTemplate from "./generated/templates/SplitButtonTemplate.lit.js";
+import ButtonDesign from "./types/ButtonDesign.js";
+
+import {
+	SPLIT_BUTTON_DESCRIPTION,
+	SPLIT_BUTTON_KEYBOARD_HINT,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Styles
+import SplitButtonCss from "./generated/themes/SplitButton.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-split-button",
+	properties: /** @lends sap.ui.webcomponents.main.SplitButton.prototype */ {
+		/**
+		 * Defines the text of the component.
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		text: {
```

Comment:
Hello @NHristov-sap the ui5-button uses the default slot for setting text, here it's via property.
Is there a technical reason for this?

Currently the text is set this way:
```html
<ui5-split-button id="sbDefault" text="Default" design="Default"></ui5-split-button>
```

perhaps it can be set this way:
```html
<ui5-split-button id="sbDefault"design="Default">Default
</ui5-split-button> (edited) 
```

The second option would make it more consistent with the ui5-button
```html
<ui5-button>Default</ui5-button>
```

Reply:
If it's done because of the MenuButton (that will be implemented in future) use-case, that would internally use a SplitButton, there should not be a difference if it's a property or slot, just the place to bind the value would be different.

if it's a property:
```hbs
<div class="menuBtnShadowRoot">
    <ui5-split-button text="{{splitBtnText}}"></ui5-split-button>
</div
```
if it's a slot:
```hbs
<div class="menuBtnShadowRoot">
    <ui5-split-button>{{splitBtnText}}</ui5-split-button>
</div
```

```js
// MenuBtn.js
// Note: the variable names are just for the sake of the explanation
metadata: {
...
     /**
     * @private
     */
     currentlySelectedMenuItemText: {
          type: String;
     }
}


get splitBtnText() {
  return this.currentlySelectedMenuItemText; // this changes upon user interaction and re-render the component
}
```

Reply:
Done.